site/content/contribute/getting-started/inactive-contribution.md

```diff
@@ -0,0 +1,24 @@
+---
+title: "Inactive Contributions"
+date: 2017-08-20T12:33:36-04:00
+weight: 1.5
+subsection: Getting Started
+---
+
+This process describes how inactive contributions are managed at Mattermost, inspired by the [Kubernetes project](https://github.com/kubernetes/kubernetes):
+
+1. After 14 days of inactivity, a contribution becomes stale.
+ - If action is required from submitter, Community Coordinator asks if the team can help clarify previous feedback or provide guidance on next steps, and adds `stale` label to the contribution.
+ - If action is required from reviewers, Community Coordinator asks reviewers to share feedback or help answer questions, and adds `stale` label to the contribution. The Coordinator will follow up with reviewers until a response is received.
+
+2. After another 14 days of inactivity, a contribution becomes rotten.
+ - Community Coordinator asks the submitter if the team can help with questions. They acknowledge that after another 30 days of inactivity the contribution will be closed. They also add a `rotten` label to the contribution.
```

Comment:
Is `rotten` is an industry standard term? It has a decidedly negative connotation to me that `stale` or `inactive` don't invoke.

Reply:
The labels were inspired by Kubernetes community: https://github.com/kubernetes/kubernetes/labels?utf8=%E2%9C%93&q=lifecycle

I'm definitely open to another term though 👍

Reply:
Honestly, I'm 0/5 on this -- if it's a standard term, I think there's value in that and I can just interpret the term differently :)

Reply:
:+1: on the negative connotation.
I'm not an native speaker and con't think of a good alternative. Maybe `orphaned` or something similar to this?

Reply:
Cool - I've heard users referring to "dead" channels/teams as `dormant` so that may be another option. Though `orphaned` is likely more familiar with non-native speakers.

So we can go with `stale` > `inactive` > `orphaned` - thoughts? (I feel that `orphaned` implies more strongly that the PR won't be worked on anymore, whereas `inactive` has the possibility of being re-activated).

Reply:
So I wonder if we could just make do with `stale` and `inactive`, closing the issue on the transition to `inactive`. It's not like that's the end: the ticket can always be reopened, but that's already 44 days of inactivity which seems long enough IMO.

Reply:
I'm not sure where do you get the 44 days from. The process is, I think,
14ht day: Add `stale`
28th day: Remove `stale` and add `inactive`
58th day: Remove `inactive`, add `orphaned` and close PR.

The main advantage of having a label just for PR's that got closed do to inactivity is, that you can search for them.

Reply:
@hanzei, so I was wondering if it made sense to just have two lifecycle labels here: `stale` after 14 days, and `inactive` (or `orphaned`) after 30 days, after which the issue would be closed. Seems like it might be easier to manage and interpret at a glance, vs. trying to remember which of `stale` and `inactive` "came first".

In the three-lifecycle-label proposal, what's the advantage of the extra label?

Reply:
The current process reaches out the community member twice (on day 14 and day 28) before closing the PR on day 58.

I'm 0/5 between a 2 or 3 lifecycle-label process - two is simpler, and I'm happy to go with that, as long as we then contact the contributor twice during the `stale` period.

--

On a side note, as we scale, I'm wondering how easy/hard it would be to add capability for [mattermod](https://github.com/mattermost/mattermost-mattermod) to automatically contact contributor and apply these labels after X number of days.

Reply:
0/5 as well, so long as we number the lifecycles, e.g. `0:stale`, `-1:orphaned`, `-2:inactive` or some such.

Love the `mattermod` idea, and honestly wouldn't be very hard! I'd say we file a ticket as part of this effort and automate it from the beginning.

Reply:
:+1: on the numbering of the lables.
:+1: on the `mattermod` idea.

I like the 3 label process because it makes the workflow clear. Every time we reached out to the contributor, we add a new "level" of label. And reaching out to the contributor twice seams is important in my opinion.

Reply:
@hanzei, framing it as the number of times we reach out to a contributor really helps clarify the proposal for me. Makes a ton of sense!

Reply:
Did we agree on calling the 2th step `orphaned`?

Reply:
I was thinking stale > orphaned > inactive. Not that we should follow Kubernetes community 1-1, but their first state is `stale` and last state is `inactive`, so I'd prefer to keep that consistent. That way contributors and users switching between the two repos have the same understanding of the inactive state.

Reply:
After think a bit about this, I would consider going with stale > inactive > orphaned. orphaned seams to me (a non native speaker) as the strongest of all three words. It also describes what happened quite well: The contributor leaved the contribution alone.

I don't think the naming of Kubernetes is so idiomatic that we must follow it. Keep in mind that we also number the steps and its quite easy to pick up the progress.